### DISCUSSION TITLE
Remove whitespaces from empty lines when pressing enter

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#24047] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Zed:
```python
def test():
    a = 1
    |                      <----cursor here, press enter
```

```python
def test():
    a = 1
    (4 whitespaces are kept on this line)
    |                      <----press enter again, cursor here, now
```

This is inconvenient when sometimes one try to run `workspace::SaveWithoutFormat` to avoid formatting but lots of the whitespaces are kept.

In Emacs and VSCode, those whitespaces are removed without needing to save the file or running formatter.

**Comment by probably-neb** [comment]
Hmm this seems tricky. Can think of lots of times this would actually be the desired behavior. Going to move this to discussions so we can keep the conversation going and think about how or if to implement this

**Comment by ivomachado** [comment]
Neovim and Helix also does like that, I hope it becomes an option or something like that.
